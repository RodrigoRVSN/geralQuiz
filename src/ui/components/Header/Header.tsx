import React from "react";
import { HeaderAppBar, HeaderLogo } from "./Header.style";
import { Container, Toolbar } from "@material-ui/core";

const Header: React.FC = () => {
  return (
    <>
      <HeaderAppBar position={"sticky"}>
        <Toolbar component={Container}>
          <HeaderLogo>GERAL QUIZ</HeaderLogo>
        </Toolbar>
      </HeaderAppBar>
    </>
  );
};

export default Header;
