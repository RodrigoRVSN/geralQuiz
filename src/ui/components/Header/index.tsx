import React from "react";
import { HeaderAppBar, HeaderLogo } from "./style";
import { Container, Toolbar } from "@material-ui/core";

const Header: React.FC = () => {
  return (
    <>
      <HeaderAppBar position={"sticky"}>
        <Toolbar component={Container}>
          <HeaderLogo><a data-testid="homePage" href="/">GERAL QUIZ</a></HeaderLogo>
        </Toolbar>
      </HeaderAppBar>
    </>
  );
};

export default Header;
