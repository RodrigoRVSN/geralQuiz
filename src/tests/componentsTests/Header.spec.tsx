import React from "react";
import { render,fireEvent,screen ,getByRole} from "@testing-library/react";
import Header from "../../ui/components/Header";

describe("<Header />", () => {

  it("should render the header", () => {
    const {getByTestId} = render(<Header />, {});
    fireEvent.click(getByTestId("homePage")); 
    expect(screen.getByRole('heading',{name: 'GERAL QUIZ'})).toBeInTheDocument();
  });

    it('should match snapshot', ()=>{
      const {container} = render(<Header/>);
      expect(container).toMatchSnapshot();
    });
});
