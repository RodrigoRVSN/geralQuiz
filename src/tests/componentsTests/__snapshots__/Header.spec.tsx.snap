// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-ij0bk7-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-19u80em-MuiContainer-root-MuiToolbar-root"
    >
      <h2
        class="css-un64pd"
      >
        <a
          data-testid="homePage"
          href="/"
        >
          GERAL QUIZ
        </a>
      </h2>
    </div>
  </header>
</div>
`;
